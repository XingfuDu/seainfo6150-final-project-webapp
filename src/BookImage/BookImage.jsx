import React from "react";
import PropTypes from "prop-types";
import classes from "./BookImage.module.css";

const BookImage = (props) => {
  return <img className={classes.image} alt={props.title} src={props.url} />;
};

BookImage.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
export default BookImage;
