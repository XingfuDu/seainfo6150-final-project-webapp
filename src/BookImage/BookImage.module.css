.image {
    border: 2px solid white;
    margin-right: 20px;
    margin-bottom: 30px;
    /* width: 20%; */
    width: 200px;
    height: 300px;
}