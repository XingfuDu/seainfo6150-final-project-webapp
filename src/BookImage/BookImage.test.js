import React from 'react';
import BookImage from './BookImage';
import { render } from "@testing-library/react";

describe('BookImage tests', () => {
	it('renders correctly', () => {
		const { container } = render(
			<BookImage
				url="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg"
				title="Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals"
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
