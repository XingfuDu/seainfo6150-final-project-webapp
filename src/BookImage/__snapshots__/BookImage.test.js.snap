// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookImage tests renders correctly 1`] = `
<div>
  <img
    alt="Some We Love, Some We Hate, Some We Eat: Why It's So Hard to Think Straight About Animals"
    class="image"
    src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1369453733l/6953508.jpg"
  />
</div>
`;
